transcom/mymove
# MB-15875 prime service item address change check

## [Jira ticket](https://dp3.atlassian.net/browse/MB-15875)

## Summary
This PR adds validations to ensure that the prime can't change an exisitng `SITDestinationFinalAddress` (prime is able to add `SITDestinationFinalAddress` if it hasn't been set yet).
 
Since `SITDestinationOriginalAddress` should only be set when approving a service item, we also added a check to make sure that `SITDestinationOriginalAddress` can't be added/updated using the updater. 

`SITDestinationOriginalAddress` isn't in the swagger definition, so there isn't really a way to test service object validation directly, but hopefully the added tests are sufficient.

Also made a small change to allow for 0 distance sit_address_updates.


## Verification Steps for the Author

These are to be checked by the author.
- [x ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. run DEVSEED_SUBSCENARIO=sit_address_updates make db_dev_e2e_populate
2. Go to a `SITUP2` and get the service item ID and etag for the `DDDSIT` service item (there should already be a `sitDestinationFinalAddress`
3. In postman, create a PATCH request to http://officelocal:3000/mto-service-items/:mtoServiceItemID with this body
```
{
  "id": {serviceItemID},
  "modelType": "UpdateMTOServiceItemSIT",
  "sitDestinationFinalAddress": {
    "streetAddress1": "7 Q St",
    "city": "San Antonio",
    "state": "TX",
    "postalCode": "78227",
  },
}
``` 
 and make sure you receive an error
4. In your db, find the service item in the `mto_service_items` table and delete the `sit_destination_final_address_id`
5. create the same PATCH request in step 3, and make sure that the service item is returned with `sitDestinationFinalAddress` populated



### Backend
